taxonomy-editor-frontend/src/pages/gotoproject/index.jsx

```diff
@@ -0,0 +1,77 @@
+import { Typography, Box, Grid } from "@mui/material";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import useFetch from "../../components/useFetch";
+import { API_URL } from "../../constants";
+import MaterialTable from '@material-table/core';
+import EditIcon from '@mui/icons-material/Edit';
+import { useEffect } from "react";
+import { toSnakeCase, toTitleCase } from "../editentry/interConvertNames";
+
+const GotoProject = () => {
+    const { data: incomingData, isPending, isError, isSuccess, errorMessage } = useFetch(`${API_URL}projects`);
+    const [projectData, setProjectData] = useState([]);
+    const navigate = useNavigate();
+
+    useEffect(() => { 
+        const renderedProjects = [];
+        if (incomingData) {
+            console.log(incomingData);
+            incomingData.forEach((element) => {
+                const projectNode = element[0];
+                renderedProjects.push({
+                    'id' : Math.random().toString(),
+                    'projectName' : projectNode['id'],
+                    'taxonomyName' : toTitleCase(projectNode['taxonomy_name']),
+                    'branchName' : projectNode['branch_name']
+                })
+            })
+        }
+        setProjectData(renderedProjects);
+    }, [incomingData]);
+
+    // Check error in fetch
+    if (isError) {
+        return (<Typography variant='h5'>{errorMessage}</Typography>)
```

Comment:
for some reason I never put attention of this, but are the errors returned from the backed user readable? can we ensure that's the case for all of the errors? if not I think it is ok to just show: 'Ooops, something went wrong :(' Maybe have a component for that case or something.

Reply:
I think it's currently just "Failed to fetch". I think we can add a component for that, or change it to "Oops. something went wrong :(".

Reply:
Think we can leave it for now as is but, I'd prefer to replace it with something generic. Don't think it makes a lot of sense to let the user if the error was one of many possible fails (server side, internet connection, etc).